kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -1327,6 +1377,121 @@ class ResidualBasedBlockBuilderAndSolver
         }
     }
 
+    /**
+     * @brief This method returns the scale norm considering for scaling the diagonal
+     * @param rA The LHS matrix
+     * @return The scale norm
+     */
+    double GetScaleNorm(TSystemMatrixType& rA)
+    {
+        if (mOptions.Is(NO_SCALING) ) {
+            return 1.0;
+        } else {
+            if (mOptions.Is(CONSIDER_PRESCRIBED_DIAGONAL)) {
+                KRATOS_DEBUG_ERROR_IF(mScaleFactor < std::numeric_limits<double>::epsilon()) << "Scale factor of the diagonal cannot be zero or almost zero" << std::endl;

```

Comment:
MINOR: this may be a KRATOS_ERROR_IF the cost is negligible also in release mode